src/coredump/coredump.c

```diff
@@ -902,7 +902,8 @@ static int submit_coredump(
 log:
         core_message = strjoina("Process ", context->meta[META_ARGV_PID],
                                 " (", context->meta[META_COMM], ") of user ",
-                                context->meta[META_ARGV_UID], " dumped core.",
+                                context->meta[META_ARGV_UID], " received a fatal signal",
+                                coredump_fd >= 0 && arg_storage != COREDUMP_STORAGE_NONE ? " and dumped core" : NULL,
```

Comment:
doesn't it need to be `""` rather than `NULL` given it's va-args, otherwise it won't process the rest? or am I getting confused with variadic macros?

Reply:
Whoops, yeah, you're right. I just got carried away by the NULLs in the following expressions.

Reply:
And just to clarify - NULLs in `__VA_ARGS__` should be fine, but we explicitly stop at the first one in `strjoina()`

https://github.com/systemd/systemd/blob/400229eaae49e3685076d1e28f82a65d471cd90c/src/basic/string-util.h#L115